src/xSTIR/pSTIR/STIR.py

```diff
@@ -1321,6 +1338,16 @@ def modality(self):
         """Returns imaging modality as Python string."""
         return parms.char_par(self.handle, 'AcquisitionData', 'modality')
 
+    @property
+    def __array_interface__(self):
+        """As per https://numpy.org/doc/stable/reference/arrays.interface.html"""
+        return {'shape': self.shape, 'typestr': '<f4', 'version': 3,
+                'data': (parms.size_t_par(self.handle, 'AcquisitionData', 'address'), False)}
+
+    def asarray(self):
+        """Returns view of self"""
+        return numpy.asarray(self)
```

Comment:
More array_api_compat...

```suggestion
    def asarray(self, xp=numpy):
        """Returns view of self"""
        return xp.asarray(self)
```

Reply:
ok

Reply:
Is there a way to expose a `view` and the older `copy`? Would help with backwards compatibility. An ugly way could be to have a bool. Not sure what the expected behaviour is of `asarray()` for non-SIRF people.